pennylane/devices/qubit/measure.py

```diff
@@ -86,17 +87,20 @@ def csr_dot_products(
         TensorLike: the result of the measurement
     """
     total_wires = len(state.shape) - is_state_batched
-    Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
-
-    if is_state_batched:
-        state = math.toarray(state).reshape(math.shape(state)[0], -1)
-
-        bra = csr_matrix(math.conj(state))
-        ket = csr_matrix(state)
-        new_bra = bra.dot(Hmat)
-        res = new_bra.multiply(ket).sum(axis=1).getA()
 
+    if is_pauli_sentence(measurementprocess.obs):
+        Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
+        state = math.toarray(state)
+        if is_state_batched:
+            state = state.reshape(math.shape(state)[0], -1)
+        else:
+            state = state.reshape(1, -1)
```

Comment:
to confirm, you're treating non-batched states as batched, but with batch_size=1? does this mean `ps.dot` now assumes all inputted states are batched, and that this measurement function will squeeze out the extra dimension before returning?

Reply:
Yes, I'm [reshaping](https://github.com/PennyLaneAI/pennylane/pull/4839/files/4d11a9407e9674db191bed8300cf223179adf0ec#diff-a289f572837e011b4483e56087180b99f765b8dadc9505e3b1ed8be55481a9f7R541) upon return, which should take care of it if I'm not mistaken.

Reply:
I'm still confused about how the issue doesn't occur, because that `vector` was [reshaped](https://github.com/PennyLaneAI/pennylane/blob/4d11a9407e9674db191bed8300cf223179adf0ec/pennylane/pauli/pauli_arithmetic.py#L532-L533) just a few lines earlier in the non-batched situation. Plus, you passed in a 1-batched vector anyway, so the reshape should still be to `(1, -1)`.

Reply:
never mind, it's because we [squeeze results before returning](https://github.com/PennyLaneAI/pennylane/blob/f424a204fab2f8b706a9df02aa7501c8dcb2a121/pennylane/devices/qubit/measure.py#L120)! makes sense.